src/chessboard/hooks/useChessGame.ts

```diff
@@ -0,0 +1,120 @@
+// @ts-nocheck
+import { useState, useEffect } from "react";
+import { Square } from "../types";
+
+type Piece = { type: string; color: "w" | "b" };
+
+// checking wheter is pawn on the last lane
+function isPawnInPromotionPosition({
+  piece,
+  targetSquare,
+}: {
+  piece: Piece;
+  targetSquare: Square;
+}): boolean {
+  if (piece?.type !== "p") return false;
+
+  const targetLine = targetSquare[1];
+
+  if (
+    (piece.color === "w" && targetLine === "8") ||
+    (piece.color === "b" && targetLine === "1")
+  ) {
+    return true;
+  }
+
+  return false;
+}
+
+export const useChessGame = (newGame) => {
```

Comment:
can this hook become part of the library?

Reply:
whilst handy for some, it removes complete control over functionality for others, it is then also dependent on chess.js and would have to maintained an upgraded as that dependency gets upgraded

the purpose of this library is just to provide the board, not the chess logic, and I also don't want to increase the bundle size too much

so taking that into account I don't think it should become part of the library, and nice idea though